app/assets/javascripts/app/views/notification_dropdown_view.js

```diff
@@ -131,6 +134,22 @@ app.views.NotificationDropdown = app.views.Base.extend({
       this.dropdownNotifications.perfectScrollbar("destroy");
       this.perfectScrollbarInitialized = false;
     }
+  },
+
+  updateHeaderCounts: function() {
+    $.getJSON(Routes.notificationsCounts(), function(data) {
+      var headerBadge = $(".notifications-link .badge"),
+        markAllReadLink = $("a#mark_all_read_link");
+      if (data.notifications > 0) {
+        headerBadge.html(parseInt(data.notifications));
+        headerBadge.removeClass("hidden");
+        markAllReadLink.removeClass("enabled");
+      } else {
+        headerBadge.addClass("hidden");
+        headerBadge.html(0);
+        markAllReadLink.removeClass("disabled");
+      }
+    });
```

Comment:
> We currently handle these kind of changes in the notification view.

https://github.com/diaspora/diaspora/blob/4364a80/app/assets/javascripts/app/views/notifications_view.js#L53

> I think it is a good idea to have them in the notification dropdown view.

Which is what you did here.

> Maybe write a helper to set the header count which you can call with the number of unread notifications.

So instead of fetching and changing the unread counts in the same function move the changes based on the unread count to a helper function which one can call with the number of unread notifications and it does all the UI changes. `updateHeaderCounts` would then just fetch the current counts and call the helper function with those.

> Then you can use this function from the notifications view.

Instead of having there some similar code you can just call your new helper function with the unread count.